### PR TITLE
Issue 1117 systemd for centos based images

### PR DESCRIPTION
**Pull request by SilinPavel**
Is related to #1117 
Configure systemd to work on centos OS systems if CP_CAP_SYSTEMD_CONTAINER parameter is specified

### CODE REVIEW

File: api/src/main/java/com/epam/pipeline/manager/execution/PipelineExecutor.java [hunk 1]
```diff
@@ -160,14 +164,18 @@ private PodSpec getPodSpec(PipelineRun run, List<EnvVar> envVars, String secretN
                 .stream()
                 .anyMatch(env -> KubernetesConstants.CP_CAP_DIND_NATIVE.equals(env.getName()) &&
                         TRUE.equals(env.getValue()));
-        spec.setVolumes(getVolumes(isDockerInDockerEnabled));
+        boolean isSystemdEnabled = ListUtils.emptyIfNull(envVars)
```
Comment (mzueva) [inline api/src/main/java/com/epam/pipeline/manager/execution/PipelineExecutor.java]: Maybe we should extract method `isParameterEnabled` and use it here and for lines 163-166?

File: api/src/main/java/com/epam/pipeline/manager/execution/PipelineExecutor.java [hunk 2]
```diff
@@ -235,7 +243,7 @@ private ContainerResources buildCpuRequests(List<EnvVar> envVars) {
                 .orElse(ContainerResources.empty());
     }
 
-    private List<Volume> getVolumes(final boolean isDockerInDockerEnabled) {
+    private List<Volume> getVolumes(final boolean isDockerInDockerEnabled, boolean isSystemdEnabled) {
```
Comment (mzueva) [inline api/src/main/java/com/epam/pipeline/manager/execution/PipelineExecutor.java]: missing `final`

File: api/src/main/java/com/epam/pipeline/manager/execution/PipelineExecutor.java [hunk 3]
```diff
@@ -246,10 +254,13 @@ private ContainerResources buildCpuRequests(List<EnvVar> envVars) {
...
             dockerMounts.forEach(mount -> volumes.add(createVolume(mount.getName(), mount.getHostPath())));
         }
+        if (isSystemdEnabled) {
+            volumes.add(createVolume(HOST_CGROUP_MOUNT.getName(), HOST_CGROUP_MOUNT.getHostPath()));
+        }
         return volumes;
     }
 
-    private List<VolumeMount> getMounts(final boolean isDockerInDockerEnabled) {
+    private List<VolumeMount> getMounts(final boolean isDockerInDockerEnabled, boolean isSystemdEnabled) {
```
Comment (mzueva) [inline api/src/main/java/com/epam/pipeline/manager/execution/PipelineExecutor.java]: missing `final`

**Review by mzueva — CHANGES_REQUESTED** [review]

**Comment by SilinPavel** [comment]
@mzueva done

**Review by mzueva — APPROVED** [review]

**Pull request merged**